recipes/jwt-cpp/all/conanfile.py

```diff
@@ -20,22 +20,33 @@ def _source_subfolder(self):
     def _supports_generic_json(self):
         return tools.Version(self.version) >= "0.5.0"
 
+    def export_sources(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            self.copy(patch["patch_file"])
+
     def requirements(self):
-        self.requires("openssl/1.1.1j")
+        self.requires("openssl/1.1.1m")
         if not self._supports_generic_json:
             self.requires("picojson/1.3.0")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version])
-        extracted_dir = self.name + "-" + self.version
-        os.rename(extracted_dir, self._source_subfolder)
+        tools.get(**self.conan_data["sources"][self.version],
+            destination=self._source_subfolder, strip_root=True)
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
```

Comment:
```suggestion

    def build(self):
        for patch in self.conan_data.get("patches", {}).get(self.version, []):
            tools.patch(**patch)
```

Reply:
I revert this a long time ago... https://github.com/prince-chrismc/conan-center-index/commit/fa8e7b1cb5b716025212e68deaf91da4f992a8a8

https://github.com/conan-io/conan-center-index/pull/1371


I remember there being a discussion around patching header only and I recall us agreeing on it should so in the source with no copy.... have we changed that?

This should probably be documented

Reply:
It's neater to patch in build() and remove no_copy_source.